### PR TITLE
User story 49

### PR DESCRIPTION
**Pull request by VinceCarollo**
## What does this PR do?

- Adds feature tests for creating items incorrectly
- Adds before validation statement to item model to check for negative values in price and inventory and turn positive
- Changes merchants/items_controller create action to not have to check for price or inventory, and to display ALL errors if form in not filled out correctly.

### CODE REVIEW

File: app/controllers/merchants/items_controller.rb [hunk 1]
```diff
@@ -10,22 +9,14 @@ def new
   end
 
   def create
-    if verify_price(params) == false
-      flash[:notice] = "The price for that item cannot be negative."
-      redirect_to dashboard_items_path
-    elsif verify_inventory_count(params) == false
-      flash[:notice] = "The inventory for that item cannot be a negative number."
+    verify_image(params)
```
Comment (BrennanAyers) [inline app/controllers/merchants/items_controller.rb:12]: Do we still need to verify the image?

Comment (VinceCarollo) [reply app/controllers/merchants/items_controller.rb]: YES!

File: app/controllers/merchants/items_controller.rb [hunk 2]
```diff
@@ -10,22 +9,14 @@ def new
...
-      verify_image(params)
-      @merchant = User.find(current_user.id)
-      @item = @merchant.items.create!(items_params)
-      if @item.save
-        flash[:notice] = "The item was created successfully."
-        redirect_to dashboard_items_path
-      else
-        render :new
-      end
+      render :new
```
Comment (BrennanAyers) [inline app/controllers/merchants/items_controller.rb:19]: 👍

File: app/models/item.rb [hunk 3]
```diff
@@ -32,4 +33,14 @@ def self.least_5_bought
...
+
+  private
+  
+  def absolute_inventory
+    write_attribute(:inventory, self.inventory.abs) unless self.inventory == nil
+  end
+
+  def absolute_price
+    write_attribute(:price, self.price.abs) unless self.price == nil
+  end
```
Comment (BrennanAyers) [inline app/models/item.rb:45]: Glad this worked!

File: app/views/merchants/items/new.html.erb [hunk 4]
```diff
@@ -17,7 +17,7 @@
 <%= f.number_field :price, min: 0 %>
 
 <%= f.label :inventory %>
-<%= f.text_field :inventory, min: 0  %>
+<%= f.number_field :inventory, min: 0  %>
```
Comment (BrennanAyers) [inline app/views/merchants/items/new.html.erb:20]: Much better!

**Review by BrennanAyers — APPROVED** [review]

**Pull request merged**